### DISCUSSION TITLE
How to run fuzzers correctly (Mac)

### DISCUSSION THREAD

**Discussion by fwh-dc** [openssl/openssl#23564] [Q&A] [answered]
I wanted to contribute fuzzers for DTLS and as a start I would like to try to build existing fuzz-targets. I looked at the readme and was able to figure out how to do compilation of fuzz-targets on macos:

```
brew install llvm@15
mkdir buildfuzz
cd buildfuzz
LDFLAGS="-L/usr/local/opt/llvm@15/lib  -Wl,-rpath,/usr/local/opt/llvm@15/lib" ../config enable-fuzz-libfuzzer \
        --with-fuzzer-lib=/usr/local/Cellar/llvm@15/15.0.7/lib/clang/15.0.7/lib/darwin/libclang_rt.fuzzer_osx.a \
        -DPEDANTIC enable-asan no-shared \
        -DFUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION \
        enable-ec_nistp_64_gcc_128 -fno-sanitize=alignment \
        enable-weak-ssl-ciphers enable-rc5 enable-md2 \
        enable-ssl3 enable-ssl3-method enable-nextprotoneg \
        --debug -lc++
```

Documentation says that fuzzers can be run with `../fuzz/helper.py x509` but that seems to statically use 'openssl/fuzz/x509' and not the build output file?

Documentation also says that I can do:
```make test TESTS=fuzz_test_asn1```
But that reports "Test fuzz_test_asn1 found no match, skipping addition...".

I can run the executable directly:
```
buildfuzz % ./fuzz/x509           
x509(35629,0x7ff848554b80) malloc: nano zone abandoned due to inability to reserve vm space.
INFO: Running with entropic power schedule (0xFF, 100).
INFO: Seed: 33187770
INFO: -max_len is not provided; libFuzzer will not generate inputs larger than 4096 bytes
INFO: A corpus is not provided, starting from an empty corpus
#2      INITED exec/s: 0 rss: 45Mb
WARNING: no interesting inputs were found so far. Is the code instrumented for coverage?
This may also happen if the target rejected all inputs we tried so far
#524288 pulse  corp: 1/1b lim: 4096 exec/s: 262144 rss: 323Mb
#1048576        pulse  corp: 1/1b lim: 4096 exec/s: 262144 rss: 535Mb
#2097152        pulse  corp: 1/1b lim: 4096 exec/s: 233016 rss: 536Mb
#4194304        pulse  corp: 1/1b lim: 4096 exec/s: 220752 rss: 536Mb
#8388608        pulse  corp: 1/1b lim: 4096 exec/s: 220752 rss: 537Mb
```
 But should I worry about the warning?

**Comment by kroeckx** [comment, marked as answer]
I always run the fuzzer manually without myself without using the helper.py file. It just adds some arguments. The important argument you're missing is that you want to point it to a directory that has a corpus in it, so ./fuzz/x509 ./fuzz/corpora/x509, or some other directory that has a corpus. Note that it will write new files to it, so you might want to create a copy of that directory if you really want to run the fuzzer. We have a fuzz-corpora repository that contains a corpus.

**Reply by fwh-dc** [reply]
Ok that's helpful. I'm not that interested in running the fuzzer. I just want to understand how it should work. Is it beneficial to add some more info to the readme? ie. how to build with macos, and how to run the fuzzer directly?
